### PR TITLE
Add ability to show address on map

### PR DESCRIPTION
**Pull request by ViktorParakhonia**

### CODE REVIEW

File: src/AppShed/Extensions/SpreadsheetBundle/Entity/Doc.php [hunk 1]
```diff
@@ -116,6 +116,33 @@ public function setUrl($url)
     }
 
     /**
+     * @var address
+     * @ORM\Column(name="address", type="string", length=355, nullable=true)
```
Comment (mcfedr) [inline src/AppShed/Extensions/SpreadsheetBundle/Entity/Doc.php:120]: this should be type="text"

File: src/AppShed/Extensions/SpreadsheetBundle/Controller/ReadController.php [hunk 2]
```diff
@@ -160,6 +184,36 @@ public function documentAction(Request $request)
         }
     }
 
+    function readGeocode($address){
```
Comment (mcfedr) [inline src/AppShed/Extensions/SpreadsheetBundle/Controller/ReadController.php:187]: This function needs to be rewritten using Guzzle.

This is mostly so that it has some proper error handling

File: src/AppShed/Extensions/SpreadsheetBundle/Controller/ReadController.php [hunk 3]
```diff
@@ -137,7 +141,27 @@ public function documentAction(Request $request)
                             $link->setScreenLink($innerScreen);
                         } else {
                             if (!empty($value)) {
-                                $innerScreen->addChild(new HTML($value));
+
+                                $address = $doc->getAddress();
```
Comment (mcfedr) [inline src/AppShed/Extensions/SpreadsheetBundle/Controller/ReadController.php:145]: why is this not outside the loop?

File: src/AppShed/Extensions/SpreadsheetBundle/Controller/ReadController.php [hunk 4]
```diff
@@ -49,6 +51,8 @@ public function indexAction(Request $request)
 
             $url = $request->get('url', false);
 
+            $address = $request->get('address', '');
```
Comment (mcfedr) [inline src/AppShed/Extensions/SpreadsheetBundle/Controller/ReadController.php:54]: default should surely be null

File: .gitignore [hunk 5]
```diff
@@ -8,3 +8,4 @@
 /bin/
 /composer.phar
 /nbproject
+/.idea
```
Comment (mcfedr) [inline .gitignore:11]: Why did you add this?

if anything you should be removing nbproject from this file

Ide/os specific ignores should be in your global ignore file https://git.ekreative.com/snippets/16

File: src/AppShed/Extensions/SpreadsheetBundle/Entity/Doc.php (hunk 1 above)
Comment (ViktorParakhonia) [inline src/AppShed/Extensions/SpreadsheetBundle/Entity/Doc.php:120]: Not - in that field we storing just name of column with real address

Comment (mcfedr) [inline src/AppShed/Extensions/SpreadsheetBundle/Entity/Doc.php:120]: ah, of course, ok, then the default 255 should be enough :)

File: src/AppShed/Extensions/SpreadsheetBundle/Controller/ReadController.php [hunk 6]
```diff
@@ -137,7 +141,27 @@ public function documentAction(Request $request)
                             $link->setScreenLink($innerScreen);
                         } else {
                             if (!empty($value)) {
-                                $innerScreen->addChild(new HTML($value));
+
+                                $address = $doc->getAddress();
+
+                                if (!empty($address) && $name == $address ) {
```
Comment (mcfedr) [inline src/AppShed/Extensions/SpreadsheetBundle/Controller/ReadController.php:147]: this empty check is pointless

File: .gitignore (hunk 5 above)
Comment (ViktorParakhonia) [inline .gitignore:11]: i using a phpStorm not NetBeans. Some people who use Netbeans after next push will push "nbproject" directory to repo if i remove it. You sure want to remove nbproject ?

File: src/AppShed/Extensions/SpreadsheetBundle/Entity/Doc.php (hunk 1 above)
Comment (ViktorParakhonia) [inline src/AppShed/Extensions/SpreadsheetBundle/Entity/Doc.php:120]: ok)

File: .gitignore (hunk 5 above)
Comment (mcfedr) [inline .gitignore:11]: i am sure you want to remove _both_ and setup your global git ignore file

Comment (ViktorParakhonia) [inline .gitignore:11]: Ok. I will try that solution.

чт, 7 мая 2015, 18:02, Fred Cox notifications@github.com:

File: src/AppShed/Extensions/SpreadsheetBundle/Controller/ReadController.php (hunk 6 above)
Comment (mcfedr) [inline src/AppShed/Extensions/SpreadsheetBundle/Controller/ReadController.php:147]: I think you can move the check to be inside the getAddress getter, and return null if its empty, this will protect against columns with empty names

Comment (ViktorParakhonia) [inline src/AppShed/Extensions/SpreadsheetBundle/Controller/ReadController.php:147]: Thanks, check will be in getter.

File: src/AppShed/Extensions/SpreadsheetBundle/Controller/ReadController.php (hunk 2 above)
Comment (mcfedr) [inline src/AppShed/Extensions/SpreadsheetBundle/Controller/ReadController.php:187]: you want to make the guzzle client as a symfony service.

for an example https://github.com/mcfedr/youtubelivestreamsbundle/blob/master/src/Mcfedr/YouTube/LiveStreamsBundle/Resources/config/services.yml#L14

Comment (ViktorParakhonia) [inline src/AppShed/Extensions/SpreadsheetBundle/Controller/ReadController.php:187]: Ok.

Comment (ViktorParakhonia) [inline src/AppShed/Extensions/SpreadsheetBundle/Controller/ReadController.php:187]: Can`t understand how works your service as Reference... So, temporary i created very simple service that drops to ReadController as argument. P.S Why we use service config in XML?

Comment (mcfedr) [inline src/AppShed/Extensions/SpreadsheetBundle/Controller/ReadController.php:187]: XML is good for you :)
